crates/starknet-server/tests/get_transaction_by_hash.rs

```diff
@@ -0,0 +1,102 @@
+            env!("CARGO_MANIFEST_DIR"),
+            "/test_data/rpc/declare_v1.json"
+        ))
+        .unwrap();
+        let declare_txn_v1: BroadcastedDeclareTransactionV1 =
+            serde_json::from_str(&json_string).unwrap();
+
+        let add_transaction = devnet
+            .json_rpc_client
+            .add_declare_transaction(starknet_rs_core::types::BroadcastedDeclareTransaction::V1(
+                declare_txn_v1.clone(),
+            ))
+            .await
+            .unwrap();
+
+        let get_transaction = devnet
+            .json_rpc_client
+            .get_transaction_by_hash(add_transaction.transaction_hash)
+            .await
+            .unwrap();
+
+        if let starknet_rs_core::types::Transaction::Declare(
+            starknet_rs_core::types::DeclareTransaction::V1(declare_v1),
+        ) = get_transaction
+        {
+            assert_eq!(
+                declare_v1.transaction_hash,
+                FieldElement::from_hex_be(DECLARE_V1_TRANSACTION_HASH).unwrap()
+            );
+        }
+    }
+
+    #[tokio::test]
+    async fn get_declere_v2_transaction_by_hash_happy_path() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+
+        // Sierra class artifact. Output of the `starknet-compile` command.
+        let path_to_cario1 =
+            concat!(env!("CARGO_MANIFEST_DIR"), r"/test_data/rpc/contract_cario_v1/output.json");
+        let contract_artifact: SierraClass =
+            serde_json::from_reader(std::fs::File::open(path_to_cario1).unwrap()).unwrap();
+
+        // Casm artifact. Output of the `starknet-sierra-compile` command.
+        let path_to_casm = concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            r"/test_data/rpc/contract_cario_v1/output-casm.json"
+        );
+        let casm_contract_definition: CompiledClass =
+            serde_json::from_reader(std::fs::File::open(path_to_casm).unwrap()).unwrap();
+        let compiled_class_hash = (casm_contract_definition.class_hash()).unwrap();
+        assert_eq!(Felt::from(compiled_class_hash).to_prefixed_hex_str(), CASM_COMPILED_CLASS_HASH);
+
+        let signer = LocalWallet::from(SigningKey::from_secret_scalar(
+            FieldElement::from_hex_be(PREDEPLOYED_ACCOUNT_PRIVATE_KEY).unwrap(),
+        ));
+        let address = FieldElement::from_hex_be(PREDEPLOYED_ACCOUNT_ADDRESS).unwrap();
+
+        let mut account =
+            SingleOwnerAccount::new(&devnet.json_rpc_client, signer, address, chain_id::TESTNET);
+        account.set_block_id(BlockId::Tag(BlockTag::Latest));
+
+        // We need to flatten the ABI into a string first
+        let flattened_class = contract_artifact.flatten().unwrap();
+        let result = account
+            .declare(Arc::new(flattened_class), compiled_class_hash)
+            .nonce(FieldElement::from_hex_be("0x0").unwrap())
+            .max_fee(FieldElement::from_hex_be("0xde0b6b3a7640000").unwrap())
```

Comment:
What is this random hex sequence? This needs two improvements:
- You can use `FieldElement::from` on a number literal
- You should add a // comment explaining why this value

Reply:
I already tried to use `FieldElement::from` from `u128` but after a call with Mario, we decided to stick with the hex value.

I added a comment to this in a97e67d81ac47215716df60836cfd4c83e6fa2e8.